core/tests/test_utils.py

```diff
@@ -1076,6 +1288,45 @@ def swap(self, obj, attr, newvalue):
         finally:
             setattr(obj, attr, original)
 
+    @contextlib.contextmanager
+    def swap_conditionally(
```

Comment:
I am rather concerned, in general, about adding this to test_utils. We generally discourage branching/logic in tests and this construct only encourages it, so I am worried that it will make tests more complex in general.

Is there a compelling need to introduce this as a general utility? If not, can we not do so?

Reply:
Nope not really, so I've removed it.